common/include/bmp180_protocol.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * baroceptor_protocol.h
+ *
+ *  Created on: 2020��10��19��
+ *      Author: duanning
+ */
+
+#ifndef BMP180_PROTOCOL_H_
+#define BMP180_PROTOCOL_H_
+
+#include "abstract_protocol.h"
+
+class Bmp180Protocol: public DummyProtocol {
+public:
+	Bmp180Protocol() {}
+	~Bmp180Protocol() {}
+
+	virtual error_condition UnpackFrame(
+			const vector<uint8_t>& buffer, vector<uint8_t>& unpacked_data) {
+		if (ExtractValidData(buffer, unpacked_data) != true) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+private:
+	virtual error_condition CalcCheckCode(\
+			const vector<uint8_t>& buffer, vector<uint8_t> check_code) {
+		if (buffer.back() != check_code.back()) {
+			return make_error_condition(errc::bad_message);
+		}
+		if (buffer.at(buffer.size() - 2) !=\
+				check_code.at(check_code.size() - 2)) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+	virtual bool ExtractValidData(const vector<uint8_t>& buffer,\
+			vector<uint8_t>& unpacked_data) {
+		if (IsFrameComplete(buffer) != true) {
+			return false;
+		}
+		for (auto i : buffer)
```

Comment:
Please remove the debug output, if we do need some debug output, please use the `Logger` class. And we need to pay attention to diffs, don't include unnecessary code.